Framework/Backend/http/server.js

```diff
@@ -242,65 +242,73 @@ class HttpServer {
   }
 
   /**
-   * Handles oAuth authentication flow (default path of the app: '/')
-   * - If query.code is valid embeds the token and grants the access to the application
-   * - Redirects to the OAuth flow if query.code is not present (origin path != /callback)
-   * - Prints out an error when code is not valid
-   * The query arguments are serialized and kept in the 'state' parameter through OAuth process
+   * Handles authentication flow (default path of the app: '/')
+   * - If JWT query.token is valid it grants the access to the application
+   * - Redirects to the OpenID flow otherwise, and sets the current state of URL
    * @param {object} req - HTTP request
    * @param {object} res - HTTP response
-   * @param {object} next - serves static paths when OAuth suceeds
-   * @return {object} redirects to OAuth flow or displays the page if JWT token is valid
+   * @param {object} next - serves static paths when OpenId suceeds
+   * @return {object} redirects to OpenID  flow or displays the page if JWT token is valid
    */
-  oAuthAuthorize(req, res, next) {
+  ident(req, res, next) {
     const query = req.query; // User's arguments
     const token = req.query.token;
 
     if (token && this.jwt.verify(token)) {
       next();
     } else {
-      // Save query params and redirect to the OAuth flow
+      // Redirects to the OpenID flow
       const state = new Buffer(JSON.stringify(query)).toString('base64');
-      return res.redirect(this.oauth.getAuthorizationUri(state));
+      return res.redirect(this.openid.getAuthUrl(state));
     }
   }
 
   /**
-   * oAuth allback route - when successfully authorized (/callback)
+   * OpenID Connect callback - when successfully authorized (/callback)
    * Redirects to the application deserializes the query parameters from state variable
    * and injects them to the url
    * @param {object} req - HTTP request
    * @param {object} res - HTTP response
-   * @return {object} redirect to address with re-included query string
    */
-  oAuthCallback(req, res) {
-    const code = req.query.code;
-    const state = req.query.state; // base64
-    if (!code || !state) {
-      return res.status(400).send('code and state required');
-    }
-
-    const query = JSON.parse(new Buffer(state, 'base64').toString('ascii'));
+  identCallback(req, res) {
+    this.openid.callback(req).then((tokenSet) => {
+      const details = tokenSet.claims();
+
+      // Set token and user details in the query
+      const query = {
+        personid: details.cern_person_id,
+        name: details.name,
+        token: this.jwt.generateToken(details.cern_person_id, details.cern_upn, this.authorise(details)),
+      };
 
-    this.oauth.createTokenAndProvideDetails(code)
-      .then((details) => {
-        query.personid = details.user.personid;
-        query.name = details.user.name;
+      // Read back user params from state
+      const userQuery = JSON.parse(new Buffer(req.query.state, 'base64').toString('ascii'));
```

Comment:
`new Buffer` is deprecated and is recommended to use `Buffer.from` or other equivalents.